src/core/Table/Table.test.tsx

```diff
@@ -205,6 +205,37 @@ it('should handle checkbox clicks', () => {
   expect(onSelect).toHaveBeenCalledWith([], expect.any(Object));
 });
 
+it('should handle row clicks', () => {
+  const onSelect = jest.fn();
+  const onRowClick = jest.fn();
+  const { container, getByText } = renderComponent({
+    isSelectable: true,
+    onSelect,
+    onRowClick,
+  });
+
+  expect(screen.queryByText('Header name')).toBeFalsy();
+  const rows = container.querySelectorAll('.iui-table-body .iui-row');
+  expect(rows.length).toBe(3);
+
+  fireEvent.click(getByText(mockedData()[1].name));
+  expect(rows[1].classList).toContain('iui-selected');
+  // expect(onSelect).toHaveBeenCalledTimes(1);
```

Comment:
This does not currently work as intended. It's called 2 times because I'm also using `instance.toggleAllRowsSelected()` to deselect all rows first. I don't think the user would expect this to be called twice when a row is only clicked once ☹️ 

I wonder if there's a way to "collect" all the events in the reducer, before firing `onSelect`.

Reply:
You could dispatch custom action:
![image](https://user-images.githubusercontent.com/36186912/123602864-e72d3a80-d801-11eb-82cf-bb26195d5d08.png)
![image](https://user-images.githubusercontent.com/36186912/123602901-f0b6a280-d801-11eb-8be1-375fa67a23f7.png)
Need to think a better name and how to handle if conditions.

`dispatch` taken from here:
![image](https://user-images.githubusercontent.com/36186912/123602987-0d52da80-d802-11eb-8070-dbcafacb9d6f.png)

Reply:
Added, but not sure about putting it in `TableActions` record.